### PR TITLE
Implementation of Github Actions 

### PR DESCRIPTION
**Pull request by celiosantosjr**
General improvements:

- Implementation of Github actions instead of Travis
  CI for continuous Macrel debugging
- Cleaning of Travis CI scripts and recipes
- Update of the badge of build status on Readme file

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: build status
+name: Build Status
    
 on:
```
Comment (luispedro) [inline .github/workflows/python-app.yml:3]: Why the `cron` job?

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -66,10 +66,9 @@ jobs:
         ./install.sh
         conda install --prefix ./envs/Macrel_env  -y pytest
 
-    - name: Activating Macrel environment
-      run: source /usr/share/miniconda3/bin/activate /home/runner/work/macrel/macrel/envs/Macrel_env
-                      
     - name: Evaluate Macrel Build
       run: |
+        source /usr/share/miniconda3/bin/activate /home/runner/work/macrel/macrel/envs/Macrel_env
+        macrel="/home/runner/work/macrel/macrel/envs/Macrel_env/bin/macrel"
```
Comment (luispedro) [inline .github/workflows/python-app.yml:72]: is this `macrel=...` necessary? it does not seem to be used elsewhere

**Review by luispedro — CHANGES_REQUESTED** [review]
See also the comments on


https://github.com/BigDataBiology/macrel/commit/7ed65cf0d864cd9fdd8488853c0f9b66aa15bf83

File: .github/workflows/python-app.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+name: Build Status
+   
+on:
+   push:
+   pull_request:
+   schedule:
+     - cron: '0 0 * * 0' # weekly
```
Comment (luispedro) [inline .github/workflows/python-app.yml:7]: Why this?

File: .github/workflows/python-app.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
```
Comment (luispedro) [inline .github/workflows/python-app.yml:18]: We don't work on macos?

File: .github/workflows/python-app.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+#        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.6, 3.7, 3.8]
+#        exclude:
```
Comment (luispedro) [inline .github/workflows/python-app.yml:21]: Cleanup these comments before commiting to `main`

File: .github/workflows/python-app.yml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            
+    - name: Checking code
+      uses: actions/checkout@v2
+  
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Display Python version
+      run: python -c "import sys; print(sys.version)"
```
Comment (luispedro) [inline .github/workflows/python-app.yml:42]: Why is this necessary?

File: .github/workflows/python-app.yml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        os: [ubuntu-latest]
+#        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.6, 3.7, 3.8]
+#        exclude:
+#          - os: macos-latest
+#            python-version: 3.8
+#          - os: windows-latest
+#            python-version: 3.6
+
+    env:
```
Comment (luispedro) [inline .github/workflows/python-app.yml:27]: Why are these necessary? If they are necessary, can we change the name to not be `TRAVIS_` ?

File: .github/workflows/python-app.yml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        activate-environment: ""
+
+
+    - name: Set ENV variables
+      run: |
+          echo "{TRAVIS_BRANCH}=${TRAVIS_BRANCH:-$(echo $GITHUB_REF | awk 'BEGIN { FS = "/" } ; { print $3 }')}" >> $GITHUB_ENV
+          echo "{TRAVIS_SECURE_ENV_VARS}=${TRAVIS_SECURE_ENV_VARS:-$(if [ -z "${{ secrets.something }}" ]; then echo "false"; else echo "true"; fi)}" >> $GITHUB_ENV
+          echo "{TRAVIS_EVENT_TYPE}=${TRAVIS_EVENT_TYPE:-$(if [ "schedule" == "${{ github.event_name }}" ]; then echo "cron"; else echo "${{ github.event_name }}"; fi)}" >> $GITHUB_ENV
+        
+    - name: Print Travis ENV vars
```
Comment (luispedro) [inline .github/workflows/python-app.yml:57]: Is this really needed? Isn't this information you know anyway?

File: .github/workflows/python-app.yml [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+          echo "TRAVIS_SECURE_ENV_VARS: ${TRAVIS_SECURE_ENV_VARS}"
+ 
+    - name: Build and test environment
+      run: |
+        ./install.sh
+        conda install --prefix ./envs/Macrel_env  -y pytest
+
+    - name: Evaluate Macrel Build
+      run: |
+        source /usr/share/miniconda3/bin/activate /home/runner/work/macrel/macrel/envs/Macrel_env
```
Comment (luispedro) [inline .github/workflows/python-app.yml:71]: Any way to avoid the hardcoded paths?

File: .github/workflows/python-app.yml [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+   
+on:
+   push:
+   pull_request:
+   schedule:
+     - cron: '0 0 * * 0' # weekly
+
+defaults:
+  run:
+    shell: bash
```
Comment (luispedro) [inline .github/workflows/python-app.yml:11]: Is `bash` not the default? why do you need `bash` specifically, please add a comment.

**Review by luispedro — CHANGES_REQUESTED** [review]

**Comment by luispedro** [comment]
> See also the comments on
> 
> [7ed65cf](https://github.com/BigDataBiology/macrel/commit/7ed65cf0d864cd9fdd8488853c0f9b66aa15bf83)

Ignore that. The issue was that that commit had been mistakenly already pushed to `master`, but I force-removed it so the comments are on this PR now.

File: .github/workflows/python-app.yml (hunk 1 above)
Comment (celiosantosjr) [reply .github/workflows/python-app.yml:3]: This is a way to make github testing once a week the build (always sunday at 00:00). We can just eliminate it if you prefer.

File: .github/workflows/python-app.yml (hunk 7 above)
Comment (celiosantosjr) [reply .github/workflows/python-app.yml]: cleaned this. it is important to import the files from github to the environment of work.

File: .github/workflows/python-app.yml (hunk 8 above)
Comment (celiosantosjr) [reply .github/workflows/python-app.yml]: eliminated this

File: .github/workflows/python-app.yml (hunk 9 above)
Comment (celiosantosjr) [reply .github/workflows/python-app.yml:44]: in Github actions is better it be hardcoded, because the other way involved using anaconda against macrel_env after configuring miniconda, so it always caused a bug.

File: .github/workflows/python-app.yml (hunk 2 above)
Comment (celiosantosjr) [reply .github/workflows/python-app.yml:45]: it is needed because the command "which" could not find it by itself for some reason when Macrel_env was activated

**Review by celiosantosjr — COMMENTED** [review]
Requested changes were reviewed. The environment previously specified was cleaned. 
The hard coded paths were kept. The schedule of testing was eliminated.

File: .github/workflows/python-app.yml [hunk 11]
```diff
@@ -0,0 +1,47 @@
+name: Build Status
+   
+on:
+   push:
+   pull_request:
+
+defaults:
+  run:
+    shell: bash # To ensure that the environment will always return to bash
```
Comment (luispedro) [inline .github/workflows/python-app.yml:9]: The question is why do you need `bash`?

**Comment by luispedro** [comment]
Closing here in favor of https://github.com/BigDataBiology/macrel/pull/34 which is a cleaned up version (removed a lot of unnecessary stuff).
